cloudbuild.yaml

```diff
@@ -23,29 +23,87 @@ steps:
       - build
       - //cmd/auth-provider-gcp
+  # build gke-exec-auth-plugin binary
+  - name: 'gcr.io/cloud-builders/bazel'
+    args:
+      - --output_user_root=/workspace/bazel-root
+      - --output_base=/workspace/bazel-base-linux-arm64
+      - build
+      - --platforms=@io_bazel_rules_go//go/toolchain:linux_arm64
+      - //cmd/gke-exec-auth-plugin
+  # build gke-gcloud-auth-plugin binary
+  - name: 'gcr.io/cloud-builders/bazel'
+    args:
+      - --output_user_root=/workspace/bazel-root
+      - --output_base=/workspace/bazel-base-linux-arm64
+      - build
+      - --platforms=@io_bazel_rules_go//go/toolchain:linux_arm64
+      - //cmd/gke-gcloud-auth-plugin
+  # build auth-provider-gcp binary
+  - name: 'gcr.io/cloud-builders/bazel'
+    args:
+      - --output_user_root=/workspace/bazel-root
+      - --output_base=/workspace/bazel-base-linux-arm64
+      - build
+      - --platforms=@io_bazel_rules_go//go/toolchain:linux_arm64
+      - //cmd/auth-provider-gcp
+  # build gke-exec-auth-plugin binary
+  - name: 'gcr.io/cloud-builders/bazel'
+    args:
+      - --output_user_root=/workspace/bazel-root
+      - --output_base=/workspace/bazel-base-windows-amd64
+      - build
+      - --platforms=@io_bazel_rules_go//go/toolchain:windows_amd64
+      - //cmd/gke-exec-auth-plugin
+  # build gke-gcloud-auth-plugin binary
+  - name: 'gcr.io/cloud-builders/bazel'
+    args:
+      - --output_user_root=/workspace/bazel-root
+      - --output_base=/workspace/bazel-base-windows-amd64
+      - build
+      - --platforms=@io_bazel_rules_go//go/toolchain:windows_amd64
+      - //cmd/gke-gcloud-auth-plugin
+  # build auth-provider-gcp binary
+  - name: 'gcr.io/cloud-builders/bazel'
+    args:
+      - --output_user_root=/workspace/bazel-root
+      - --output_base=/workspace/bazel-base-windows-amd64
+      - build
+      - --platforms=@io_bazel_rules_go//go/toolchain:windows_amd64
+      - //cmd/auth-provider-gcp
 # upload auth-provider-gcp. gke-gcloud-auth-plugin and gke-exec-auth-plugin binaries
 # TODO: figure out how to do this better, most probably getting rid of bazel
 artifacts:
   objects:
-    location: 'gs://k8s-staging-cloud-provider-gcp/auth-linux-amd64-$_PULL_BASE_REF'
+    location: 'gs://k8s-staging-cloud-provider-gcp/cloud-auth-provider-$_GIT_TAG'
     paths:
-      - '/workspace/plugin-binaries/execroot/io_k8s_cloud_provider_gcp/bazel-out/k8-fastbuild/bin/cmd/auth-provider-gcp/auth-provider-gcp_/auth-provider-gcp'
-      - '/workspace/plugin-binaries/execroot/io_k8s_cloud_provider_gcp/bazel-out/k8-fastbuild/bin/cmd/gke-gcloud-auth-plugin/gke-gcloud-auth-plugin_/gke-gcloud-auth-plugin'
-      - '/workspace/plugin-binaries/execroot/io_k8s_cloud_provider_gcp/bazel-out/k8-fastbuild/bin/cmd/gke-exec-auth-plugin/gke-exec-auth-plugin_/gke-exec-auth-plugin'
+      - '/workspace/bazel-base-linux-amd64/execroot/io_k8s_cloud_provider_gcp/bazel-out/k8-fastbuild/bin/cmd/auth-provider-gcp/auth-provider-gcp_/auth-provider-gcp'
```

Comment:
the arm64 and amd64 binaries have the same name and ones seems to override the others @ndixita 

https://console.cloud.google.com/storage/browser/k8s-staging-cloud-provider-gcp/cloud-auth-provider-v20230601-ccmv27.1.2-2-g30b952b6?pageState=(%22StorageObjectListTable%22:(%22f%22:%22%255B%255D%22))&project=k8s-staging-cloud-provider-gcp&supportedpurview=project&prefix=&forceOnObjectsSortingFiltering=false

Reply:
looking into it

Reply:
@aojea do you think I could add a command to rename the files in the same step? It is a little hacky though. Do you have any other suggestions?

cc: @BenTheElder